src/macros.rs

```diff
@@ -173,6 +173,11 @@ macro_rules! redis_module {
             argv: *mut *mut $crate::raw::RedisModuleString,
             argc: std::os::raw::c_int,
         ) -> std::os::raw::c_int {
+            unsafe {
+                // We trust that Redis won't give us a null pointer as
+                // the module context pointer.
+                $crate::MODULE_CONTEXT = std::ptr::NonNull::new(ctx);
```

Comment:
This ctx here is located on the stack and will be gone once the OnLoad function will finished. We should create a detached context here and use it instead. Or better, why not give the logger `setup` function the detached context that should be created by the user?